### PR TITLE
(SIMP-3115) Added wrapper stages to simplib

### PR DESCRIPTION
**Pull request by trevor-vaughan**
There have been several situations where various SIMP components really
needed to be wrapped in a Puppet stage to protect the rest of the
system from a particularly aggressive action (usually one that requires
a reboot).

This change adds a couple of SIMP stages that wrap the stdlib stages to
provide a buffer zone for SIMP-specific activities. This is not a
perfect solution but it will work better than just trying to tag onto
the existing stages that don't actually map to our use cases.

SIMP-3115 #comment Added buffer stages for module locking

### CODE REVIEW

File: lib/facter/simplib_sysctl.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# _Description_
```
Comment (jeefberkey) [inline lib/facter/simplib_sysctl.rb]: @trevor-vaughan you said you had a better way to do this. It should go into another PR and ticket.

Comment (trevor-vaughan) [inline lib/facter/simplib_sysctl.rb]: This *is* the better way.

The *other* way is to dump all of sysctl into a hash, but I don't think we want that madness in every `facter` run.
